projects/collection-editor-library/src/lib/services/editor/editor.service.ts

```diff
@@ -530,21 +532,37 @@ export class EditorService {
 
   getBranchingLogicByFolder(identifier) {
     const nodeData = this.treeService.getNodeById(identifier);
+    this.selectedPrimaryCategory=_.get(nodeData,'data.primaryCategory');
```

Comment:
Why are we setting the primary category in a method which has a specific purpose?